api/v1beta1/openstackdataplanenodeset_types.go

```diff
@@ -190,17 +198,55 @@ func (instance OpenStackDataPlaneNodeSet) GetAnsibleEESpec() AnsibleEESpec {
 	}
 }
 
-// DataplaneAnsibleImageDefaults default images for dataplane services
-type DataplaneAnsibleImageDefaults struct {
-	Frr                        string
-	IscsiD                     string
-	Logrotate                  string
-	NeutronMetadataAgent       string
-	NeutronSRIOVAgent          string
-	NovaCompute                string
-	OvnControllerAgent         string
-	OvnBgpAgent                string
-	TelemetryCeilometerCompute string
-	TelemetryCeilometerIpmi    string
-	TelemetryNodeExporter      string
+const (
+	// FrrDefaultImage -
+	FrrDefaultImage = "quay.io/podified-antelope-centos9/openstack-frr:current-podified"
+	// IscsiDDefaultImage -
+	IscsiDDefaultImage = "quay.io/podified-antelope-centos9/openstack-iscsid:current-podified"
+	// LogrotateDefaultImage -
+	LogrotateCrondDefaultImage = "quay.io/podified-antelope-centos9/openstack-cron:current-podified"
+	// NeutronMetadataAgentDefaultImage -
+	NeutronMetadataAgentDefaultImage = "quay.io/podified-antelope-centos9/openstack-neutron-metadata-agent-ovn:current-podified"
+	// NeutronSRIOVAgentDefaultImage -
+	NeutronSRIOVAgentDefaultImage = "quay.io/podified-antelope-centos9/openstack-neutron-sriov-agent:current-podified"
+	// NovaComputeDefaultImage -
+	NovaComputeDefaultImage = "quay.io/podified-antelope-centos9/openstack-nova-compute:current-podified"
+	// OvnControllerDefaultImage -
+	OvnControllerDefaultImage = "quay.io/podified-antelope-centos9/openstack-ovn-controller:current-podified"
+	// OvnBgpAgentDefaultImage -
+	OvnBgpAgentDefaultImage = "quay.io/podified-antelope-centos9/openstack-ovn-bgp-agent:current-podified"
+	// TelemetryCeilometerComputeImage -
+	TelemetryCeilometerComputeImage = "quay.io/podified-antelope-centos9/openstack-telemetry-ceilometer-compute:current-podified"
+	// TelemetryCeilometerIpmiImage -
+	TelemetryCeilometerIpmiImage = "quay.io/podified-antelope-centos9/openstack-telemetry-ceilometer-ipmi:current-podified"
+	// TelemetryNodeExporterImage -
+	TelemetryNodeExporterImage = "quay.io/podified-antelope-centos9/openstack-telemetry-node-exporter:current-podified"
+)
+
+var ContainerImageDefaults openstackv1.ContainerImages
+
+// SetupDefaults - initializes any CRD field defaults based on environment variables
+// called from main.go
+func SetupDefaults() {
+	// Acquire environmental defaults and initialize dataplane defaults with them
+	ContainerImageDefaults = openstackv1.ContainerImages{
+		ContainerTemplate: openstackv1.ContainerTemplate{
```

Comment:
I would have expected that I could use the ContainerTemplate fields directly in the ContainerImages struct since ContainerTemplate is embedded, but that did not work.

Reply:
nope, this is the way to initialize the embedded fields. The fields will be accessible directly on the ContainerImage though.